easybuild/easyblocks/generic/tarball.py

```diff
@@ -57,8 +75,25 @@ def build_step(self):
 
     def install_step(self, src=None):
         """Install by copying from specified source directory (or 'start_dir' if not specified)."""
-        remove_dir(self.installdir)
-        copy_dir(src or self.cfg['start_dir'], self.installdir, symlinks=self.cfg['keepsymlinks'])
+
+        # Run preinstallopts before copy of source directory
+        if self.cfg['preinstallopts']:
+            preinstall_opts = self.cfg['preinstallopts'].split('&&')
+            preinstall_cmd = '&&'.join([opt for opt in preinstall_opts if opt and not opt.isspace()])
+            self.log.info("Preparing installation of %s using command '%s'..." % (self.name, preinstall_cmd))
+            run_cmd(preinstall_cmd, log_all=True, simple=True)
+
+        # Copy source directory
+        source_path = src or self.cfg['start_dir']
+
+        if 'final_dir' in self.cfg and self.cfg['final_dir']:
```

Comment:
I wonder if we need to be giving the user the power to name the installation subdirectory (and if we do we should check they're trying to pass an absolute path)? Instead, we could create an `install_in_subdir` boolean and use the name of the software as the subdir, this would lead to more predictable results (that we would be able to leverage somewhere else without parsing the associated easyconfig).

Reply:
I agree :+1:  it is much safer to set the name of the subdir to the software name and it is a suitable solution for the easyconfig that triggered these changes.